content/en/cd-as-a-service/concepts/deployment/how-hpa-works.md

```diff
@@ -0,0 +1,73 @@
+---
+title: "How CD-as-a-Service implements Kubernetes HPA"
+linktitle: "HPA"
+description: >
+  Armory CD-as-a-Service supports Kubernetes deployments with HPA.
+---
+ ## What is HPA?
+ Kubernetes Horizontal Pod Autoscaling (HPA) scales deployments in reaction to changes in observed metrics. When the deployment load increases, the number of deployed pods is increased to meet the demand. The scale behavior is horizontal, rather than vertical, which means that rather than assigning an existing pod more memory or CPU to handle the incoming traffic, new pods that identical instances of your application, are created.
+
+ An HPA resource has three components:
+- A set of one or more metrics, each with a desired target value.
+- A reference to a Kubernetes resource, typically a Deployment but any resource that supports the Scale API sub-resource (e.g., `ReplicaSet`, `StatefulSet`). The HPA scales this resource in response to changes in observed metrics.
+- A minimum and maximum number of replicas.
+
+HPA checks metrics as configured in the deployment file AT A DEFAULT 30 SEC intervals. When the threshold load is met the HPA increases or decreases the number of pods. 
+ 
+ Kubernetes implements Horizontal Auto Scaling (HPA) as a control loop that runs intermittently. `HorizontalPodAutoscaler` controls the scale of a Deployment and its `ReplicaSet`. 
+ 
+ HPA is not recommended for replication controllers. 
+ 
+ HPA does not manage rolling updates by manipulating the replication controller. The deployment object manages the size of underlying replica sets. 
+
+ The sync interval is set by the `kube-controller-manager`. The default interval is 30 seconds. Change the default interval using the `horizontal-pod-autoscaler-sync-period` flag.
+
+The default delay is 3 minutes after an upscale event to allow metrics to stabilize. Change the default upscale delay period using the `horizontal-pod-autoscaler-upscale-delay` flag.
+ 
+## How HPA works in CD-as-a-Service
+When HPA is configured for an Armory Continuous Deployment-as-a-Service (CD-as-a-Service) deployment the deployment is converted to a `ReplicaSet` and the HPA configuration file is automatically re-written to reference the generated `ReplicaSet` resource. Continuous Deployment-as-a-Service freezes scaling behavior during a deployment.
+
+Consider a deployment to upgrade from the v1 to v2 version of your application. In this scenario the v1 application is running with 10 replicas and is scaled by HPA (min: 5 pods, max: 15 pods).
+ - At the beginning of a deployment, the HPA is deleted. Your v1 application has 10 pods but no longer scales between 5 and 15 pods.
+ - CD as a Service deploys the v2 instance of your application with 10 pods (20 total pods between v1 and v2).
```

Comment:
```suggestion
 - CD-as-a-Service deploys the `v2` instance of your application with 10 pods (20 total pods between `v1` and `v2`).
```

Reply:
fixed the product name consistency throughout during re-write

Reply:
will codify v1/v2 in new commit